tests/test_z_async.py

```diff
@@ -0,0 +1,59 @@
+##################################################################
+# Copyright 2018 Open Source Geospatial Foundation and others    #
+# licensed under MIT, Please consult LICENSE.txt for details     #
+##################################################################
+
+import unittest
+import time
+from pywps import Service, configuration
+from pywps import get_ElementMakerForVersion
+from pywps.tests import client_for, assert_response_accepted
+from .processes import Sleep
+from owslib.wps import WPSExecution
+from pathlib import Path
+
+VERSION = "1.0.0"
+
+WPS, OWS = get_ElementMakerForVersion(VERSION)
+
+
+class ExecuteTest(unittest.TestCase):
+    def setUp(self) -> None:
+        configuration.CONFIG.set('processing', 'mode', 'distributed')
+
+    def tearDown(self) -> None:
+        configuration.load_configuration()
+
+    def test_async(self):
+        client = client_for(Service(processes=[Sleep()]))
+        wps = WPSExecution()
+
+        # Build an asynchronous request (requires specifying outputs and setting the mode).
+        doc = wps.buildRequest('sleep',
+                               inputs=[('seconds', '.01')],
+                               output=[('finished', None, None)],
+                               mode='async')
+
+        resp = client.post_xml(doc=doc)
+        wps.parseResponse(resp.xml)
+        assert_response_accepted(resp)
+
+        # Wait for process to complete. The test will fail otherwise, which confirms the process is asynchronous.
+        time.sleep(.5)
```

Comment:
The check should be done explicitly to validate that the current status is still running with a new `checkStatus`, but is not yet `ProcessSucceeded`. 

If the process is somehow running in sync, this would not confirm that async was applied. It would just do a small wait before checking the status of an already completed process.